modyn/tests/supervisor/internal/test_supervisor.py

```diff
@@ -438,9 +438,18 @@ def test_get_pipeline_status_exit(
     # mock_start.side_effect = noop
     # with patch("multiprocessing.Process.start", mock_start):
 
+    start_time = time.time()
     res = sup.start_pipeline(pipeline_config, EVALUATION_DIRECTORY)
-    time.sleep(1)
-    msg = sup.get_pipeline_status(res["pipeline_id"])
+
+    # iteratively query the pipeline status until it is no longer running
+    while time.time() - start_time < 30:
+        msg = sup.get_pipeline_status(res["pipeline_id"])
+        if msg["status"] != PipelineStatus.RUNNING:
+            break
+        time.sleep(2)
+    else:
+        raise TimeoutError("Pipeline did not finish in time")
+
```

Comment:
On my local computer, the original test fails because sleeping 1 second is not enough for the pipeline to finish (it is still running after 1s). I made an iterative loop to only check the pipeline status when it finishes running.

Reply:
Thanks for cleaning all of this up, Xianzhe. Much appreciated!